ci/scripts/cpp_test.sh

```diff
@@ -114,6 +120,41 @@ else
     "$@"
 fi
 
+# This is for testing find_package(Arrow).
+#
+# Note that this is not a perfect solution. We should improve this
+# later.
```

Comment:
It seems that this new check is breaking some C++ builds (CUDA, Thread Sanitizer). @kou

Reply:
Oh, sorry. I missed these failures: https://github.com/apache/arrow/issues/47840